third_party/move/move-vm/runtime/src/data_cache.rs

```diff
@@ -157,7 +156,9 @@ impl<'r> TransactionDataCache<'r> {
             let mut resources = BTreeMap::new();
             for (ty, (layout, gv, has_aggregator_lifting)) in account_data_cache.data_map {
                 if let Some(op) = gv.into_effect_with_layout(layout) {
-                    let struct_tag = match loader.type_to_type_tag(&ty, module_storage)? {
+                    let ty_tag_builder =
+                        TypeTagConverter::new(module_storage.runtime_environment());
+                    let struct_tag = match ty_tag_builder.ty_to_ty_tag(&ty)? {
```

Comment:
can't we pass module_storage.runtime_environment as a parameter to ty_to_ty_tag and avoid creating the converter?

Reply:
Converter is just an encapsulation of conversion logic? Do you prefer having this a non-class function `fn ty_to_ty_tag(ty: &Type, env: &RuntimeEnvironment) -> ... `?

Reply:
yeah, I meant a free function but member of environment could be even better. Was a suggestion to see if it could look slightly better - feel free to do as seems nicer to you.